src/lex.ll

```diff
@@ -504,9 +509,15 @@ L?\"(\\.|[^\\"])*\" { lStringConst(&yylval, &yylloc); return TOKEN_STRING_LITERA
     return TOKEN_DOUBLE_CONSTANT;
 }
 
-
 {FLOAT_NUMBER} {
     RT;
+    std::string val(yytext);
+    std::string fp16("f16");
+    if (val.size() >= fp16.size() && ((val.compare(val.size() - fp16.size(), fp16.size(), fp16) == 0)
+           || (val.compare(val.size() - fp16.size(), fp16.size(), "F16") == 0))) {
```

Comment:
Let's do `f16` and `F16` checks uniformly - create a `F16` string as well.
It's equivalent, but uniformity is a good thing in the source code.

Reply:
Done